app/client/ui/DocMenu.ts

```diff
@@ -529,6 +533,16 @@ export function makeDocOptionsMenu(home: HomeModel, doc: Document, renaming: Obs
       dom.cls('disabled', !roles.canEdit(orgAccess)),
       testId('pin-doc')
     ),
+    menuItem(() => {
+        const {appModel} = page;
+        if (!appModel.currentValidUser) {
+          page.clearUnsavedChanges();
```

Comment:
This code is not needed, is it? DocPageModel should be only created if we've opened a document. Here we are somehow "lucky" that this `constructor` isn't doing anything destructive, and is disposed when we open a document (thanks to the internal implementation in `gristUrlState.ts/needPageLoad`). But still it does create some listeners and computed properties (that do something when url is changed), which is not ideal, and can lead to unexpected bugs later - as, probably, the absence of `doc` in `DocPageModel` is rather perceived as temporary state, but here it is enforced.

I think the main culprit here is that `clearUnsavedChanges` is hold in the `DocPageModel` and not on the `AppModel` level, the ideal situation for me would be to refactor it:
- Lift the `clearUnsavedChanges` method to the `AppModel` or `TopAppModel`
- Don't create the `DocPageModelImpl` at all
- Refactor `makeCopy` so that uses only `AppModel` instance (I looked into its code, and it only needs this `clearUnsavedChanges` method